### PR TITLE
Add DeviceBase class api check to test_psu.py

### PR DESCRIPTION
**Pull request by sujinmkang**
Add DeviceBase class api check to test_psu.py

### CODE REVIEW

File: tests/platform_tests/api/test_psu.py [hunk 1]
```diff
@@ -35,6 +35,49 @@ def setup(self, platform_api_conn):
...
 
+    #
+    # Functions to test methods inherited from DeviceBase class
+    #
+
+    def test_get_name(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_psus):
+            name = psu.get_name(platform_api_conn, i)
+            if self.expect(name is not None, "Unable to retrieve transceiver {} name".format(i)):
+                self.expect(isinstance(name, str), "Transceiver {} name appears incorrect".format(i))
```
Comment (jleveque) [inline tests/platform_tests/api/test_psu.py:46]: s/transceiver/PSU/

Please do this throughout the file

**Comment by dipali0125** [comment]
Is it required to validate the platform with the new platform test(tests/platform_tests/api/*), are the old tests (e.g.: tests/platform_tests/test_platform_info.py)going to become obsolete?

**Comment by jleveque** [comment]
@dipali0125:

> Is it required to validate the platform with the new platform test(tests/platform_tests/api/*)

Yes. These tests are meant to ensure the platform API is fully and properly implemented on a given platform.

>are the old tests (e.g.: tests/platform_tests/test_platform_info.py)going to become obsolete?

Some tests in that file may continue to be used, but will probably be moved to become part of another suite of tests.

**Comment by dipali0125** [comment]
Thanks for your prompt response. Few clarifications:
Are these supported in pmon2.0 or also in older pmon?
Looking at these tests , they look more like unit tests to verify the platform apis . Is there a specific reason why the new test infrastructure was added using http requests rather than using the cli commands? 
Can we modify the exisiting tests to add verification for DUT specific info like number of fans expected, no. of psu expected?

**Comment by jleveque** [comment]
> Are these supported in pmon2.0 or also in older pmon?
> Looking at these tests , they look more like unit tests to verify the platform apis . Is there a specific reason why the new test infrastructure was added using http requests rather than using the cli commands?

These tests are meant to verify that the vendor has fully and properly implemented the platform API 2.0 for the platform in question. Using the HTTP server to hook into the API itself is a clean way to test all API calls.

> Can we modify the exisiting tests to add verification for DUT specific info like number of fans expected, no. of psu expected?

Yes. This work is in progress. We are planning for the platform.json file to specify all the known truths for the platform (like number of PSUs, fans, etc.). Please see: https://github.com/Azure/sonic-buildimage/pull/5189

**Pull request merged**